packages/ember-data/lib/system/model/errors.js

```diff
@@ -91,19 +91,8 @@ var MapWithDefault = Ember.MapWithDefault;
   @uses Ember.Enumerable
   @uses Ember.Evented
  */
-export default Ember.ArrayProxy.extend(Ember.Evented, {
-  /**
-    Register with target handler
+export default Ember.ArrayProxy.extend({
 
-    @method registerHandlers
-    @param {Object} target
-    @param {Function} becameInvalid
-    @param {Function} becameValid
-  */
-  registerHandlers: function(target, becameInvalid, becameValid) {
-    this.on('becameInvalid', target, becameInvalid);
-    this.on('becameValid', target, becameValid);
-  },
```

Comment:
The behivor is not dicumented but it is definitly used by some applications. Basicly the use case is to be able to implement client side validation in terms of `errors` object. Pushing an error in to `errors` object transitions the record in to invalid state.

I am fine with deprecating it and removing in next major version. But I am not sure where the deprecation should be. For example in my app, I don't care about events at all. What I do rely on, is the fact that if I am pushing an error in to `errors` object, the record became invalid.